backend/node_engine/email_nodes.py

```
@@ -1,678 +1,588 @@
-                    return False
-            if "before" in date_criteria:
-                before_date = datetime.fromisoformat(date_criteria["before"].replace("Z", "+00:00"))
-                if email_date > before_date:
+        # Check date range
+        if "date_from" in criteria:
+            try:
+                email_date = datetime.fromisoformat(email.get("date", "").replace('Z', '+00:00'))
+                from_date = datetime.fromisoformat(criteria["date_from"])
+                if email_date < from_date:
                     return False
+            except:
```

Ruff (0.14.2)

Do not use bare `except`

(E722)